grace/run.py

```diff
@@ -57,57 +57,31 @@ def run_grace(config_file: Union[str, os.PathLike]) -> None:
         subfolder_path = run_dir / subfolder
         subfolder_path.mkdir(parents=True, exist_ok=True)
 
-    # Process the datasets as desired:
-    def prepare_dataset(
-        image_dir: Union[str, os.PathLike],
-        grace_dir: Union[str, os.PathLike],
-        num_hops: int | str,
-        connection: str = "spiderweb",
-        verbose: bool = True,
-    ) -> tuple[list]:
-        # Read the data & terate through images & extract node features:
-        input_data = ImageGraphDataset(
-            image_dir=image_dir,
-            grace_dir=grace_dir,
-            image_filetype=config.filetype,
-            keep_node_unknown_labels=config.keep_node_unknown_labels,
-            keep_edge_unknown_labels=config.keep_edge_unknown_labels,
-        )
-
-        # Process the (sub)graph data into torch_geometric dataset:
-        target_list, subgraph_dataset = [], []
-        desc = "Extracting patch features from images... "
-        for _, target in tqdm(input_data, desc=desc, disable=not verbose):
-            file_name = target["metadata"]["image_filename"]
-            LOGGER.info(f"Processing file: {file_name}")
-
-            # Store the valid graph list:
-            target_list.append(target)
-
-            # Chop graph into subgraphs & store:
-            graph_data = dataset_from_graph(
-                target["graph"],
-                num_hops=num_hops,
-                connection=connection,
-            )
-            subgraph_dataset.extend(graph_data)
-
-        return target_list, subgraph_dataset
+    # Create a transform function with frozen arguments:
+    check_and_chop_partial = partial(
```

Comment:
I find the name of this function a bit confusing, it is not very clear to what is happening here. As is in the main run.py scrip maybe we should add more comments to explain what is happening?